apps/web/modules/survey/follow-ups/lib/follow-ups.ts

```diff
@@ -0,0 +1,269 @@
+
+  // Check if 'to' is a direct email address (team member or user email)
+  const parsedEmailTo = z.string().email().safeParse(to);
+  if (parsedEmailTo.success) {
+    try {
+      // 'to' is a valid email address, send email directly
+      await sendFollowUpEmail({
+        html: body,
+        subject,
+        to: parsedEmailTo.data,
+        replyTo,
+        survey,
+        response,
+        attachResponseData: properties.attachResponseData,
+        logoUrl,
+      });
+
+      return {
+        followUpId,
+        status: "success" as const,
+      };
+    } catch (error) {
+      return {
+        followUpId,
+        status: "error",
+        error: error instanceof Error ? error.message : "Something went wrong",
+      };
+    }
+  }
+
+  // If not a direct email, check if it's a question ID or hidden field ID
+  if (!toValueFromResponse) {
+    // throw new Error(`"To" value not found in response data for followup: ${followUpId}`);
+    return {
+      followUpId,
+      status: "error",
+      error: `To value not found in response data for followup: ${followUpId}`,
+    };
+  }
+
+  if (typeof toValueFromResponse === "string") {
+    // parse this string to check for an email:
+    const parsedResult = z.string().email().safeParse(toValueFromResponse);
+    if (parsedResult.data) {
+      // send email to this email address
+      await sendFollowUpEmail({
+        html: body,
+        subject,
+        to: parsedResult.data,
+        replyTo,
+        logoUrl,
+        survey,
+        response,
+        attachResponseData: properties.attachResponseData,
+      });
+
+      return {
+        followUpId,
+        status: "success" as const,
+      };
+    } else {
+      return {
+        followUpId,
+        status: "error",
+        error: `Email address is not valid for followup: ${followUpId}`,
+      };
+    }
+  } else if (Array.isArray(toValueFromResponse)) {
+    const emailAddress = toValueFromResponse[2];
+    if (!emailAddress) {
+      return {
+        followUpId,
+        status: "error",
+        error: `Email address not found in response data for followup: ${followUpId}`,
+      };
+    }
+
+    const parsedResult = z.string().email().safeParse(emailAddress);
+    if (parsedResult.data) {
+      await sendFollowUpEmail({
+        html: body,
+        subject,
+        to: parsedResult.data,
+        replyTo,
+        logoUrl,
+        survey,
+        response,
+        attachResponseData: properties.attachResponseData,
+      });
+
+      return {
+        followUpId,
+        status: "success" as const,
+      };
+    } else {
+      return {
+        followUpId,
+        status: "error",
+        error: `Email address is not valid for followup: ${followUpId}`,
+      };
+    }
+  }
+
+  return {
+    followUpId,
+    status: "error",
+    error: "Something went wrong",
+  };
+};
+
+export const sendSurveyFollowUps = async (
+  organizationId: string,
+  surveyId: string,
```

Comment:
since we are getting the survey either way, could we tighten this more by deriving the organizationId from the surveyId?
Or even get everything from the responseId.